pkgs/top-level/all-packages.nix

```diff
@@ -1345,7 +1345,18 @@ with pkgs;
 
   substituteAllFiles = callPackage ../build-support/substitute-files/substitute-all-files.nix { };
 
-  replaceDependency = callPackage ../build-support/replace-dependency.nix { };
+  replaceDependencies = callPackage ../build-support/replace-dependencies.nix { };
+
+  replaceDependency = { drv, oldDependency, newDependency, verbose ? true }: replaceDependencies {
+    inherit drv verbose;
+    replacements = [{
+      inherit oldDependency newDependency;
+    }];
+    # When newDependency depends on drv, instead of causing infinite recursion, keep it as is.
+    cutoffPackages = [ newDependency ];
```

Comment:
As mentioned in the other comment, should the `cutoffPackages` in general default to `map (x: x.newDependency) replacements`?

Reply:
What is a disadvantage in the general case is not applicable here, so keeping backwards compatibility has little cost. If we don't want the difference, then I'd rather remove this here because it only handles a very weird edge case that I assume to be mostly irrelevant in practuce.